### PR TITLE
Additional go vet fixes

### PR DESCRIPTION
**Pull request by goltermann**
Mostly:
- pass lock by value
- bad syntax for struct tag value
- example functions not formatted properly

### CODE REVIEW

**Comment by goltermann** [comment]
Additional fixes for #22523

File: pkg/runtime/embedded.go [hunk 1]
```diff
@@ -24,9 +24,9 @@ import (
 )
 
 type encodable struct {
-	e        Encoder `json:"-"`
+	E        Encoder `json:"-"`
 	obj      Object
-	versions []unversioned.GroupVersion `json:"-"`
+	Versions []unversioned.GroupVersion `json:"-"`
```
Comment (krousey) [inline pkg/runtime/embedded.go]: This change is curious. Was vet complaining about json tags on private members? I think if that's the case, remove the tags and leave the members private.

File: test/e2e/util.go [hunk 2]
```diff
@@ -312,18 +312,19 @@ func logf(level string, format string, args ...interface{}) {
 }
 
 func Logf(format string, args ...interface{}) {
-	logf("INFO", format, args...)
+	msg := fmt.Sprintf(format, args...)
+	logf("INFO %v", msg)
```
Comment (krousey) [inline test/e2e/util.go]: These changes are wrong. The first parameter of `logf` is a level, not a format string. Unfortunately go vet bases the formatting string analysis off of function names. https://golang.org/cmd/vet/#hdr-Printf_family

I would "fix" this by renaming `logf` to `log`.

**Comment by krousey** [comment]
Most of these look fine. 

I'd like @bgrant0607 to look at the API changes. I think these actually unbreak unintended consequences, but up to him since I think we've technically released them.

I'd also like @karlkfi to look at the mesos changes.

File: pkg/runtime/embedded.go (hunk 1 above)
Comment (goltermann) [inline pkg/runtime/embedded.go]: This was "Line 29: error: struct field versions has json tag but is not exported (vet)"

The better fix is to remove the quotes around the - ?

File: test/e2e/util.go (hunk 2 above)
Comment (goltermann) [inline test/e2e/util.go]: Line 310 (just above here) defines logf.  Do you want me to create a similar function called log that does an FPrint to GingkoWriter?

Or call logf and pass an extra "%v" as a param to it?

File: pkg/runtime/embedded.go (hunk 1 above)
Comment (krousey) [inline pkg/runtime/embedded.go]: No. Just remove the json tag completely. Those tags are trying to tell the JSON package not to marshal or unmarshal those fields, but the JSON package already ignores private members.

File: test/e2e/util.go (hunk 2 above)
Comment (krousey) [inline test/e2e/util.go]: Just rename the function `logf` to `log`.

**Comment by karlkfi** [comment]
@jdef punting the mesos change review to you.

**Comment by bgrant0607** [comment]
go vet doesn't run in our build/test by default?

File: pkg/api/v1/types.go [hunk 3]
```diff
@@ -2048,7 +2048,7 @@ type NodeSpec struct {
 // DaemonEndpoint contains information about a single Daemon endpoint.
 type DaemonEndpoint struct {
 	// Port number of the given endpoint.
-	Port int32 `json:port`
+	Port int32 `json:"port"`
```
Comment (bgrant0607) [inline pkg/api/v1/types.go]: Ugh.

@gmarek Is this API in use anywhere?

We may need to ask on kubernetes-dev whether this breaks any clients.

Comment (bgrant0607) [inline pkg/api/v1/types.go]: Looks like it is used, here:
https://github.com/kubernetes/kubernetes/blob/d2d349bc84dd3a0aa8b220a8cbbbb8b240dc516d/pkg/registry/node/strategy.go#L183

So, we can't fix this, unfortunately.

Comment (gmarek) [inline pkg/api/v1/types.go]: Answering your question - it is, API server uses info in this when setting up proxy to the Node (which means proxy to the kubelet in practice).

Comment (goltermann) [inline pkg/api/v1/types.go]: No, go vet doesn't run by default.  I'll be adding it via #22524, once I'm done with all these cleanups.

File: test/e2e/util.go (hunk 2 above)
Comment (goltermann) [inline test/e2e/util.go]: Fixed, and that change meant I had to update example_k8spetstore.go as well.

File: test/e2e/util.go [hunk 4]
```diff
@@ -307,23 +307,23 @@ func nowStamp() string {
...
 
 func Logf(format string, args ...interface{}) {
-	logf("INFO", format, args...)
+	log("INFO", format, args...)
 }
 
 func Failf(format string, args ...interface{}) {
 	msg := fmt.Sprintf(format, args...)
-	logf("FAIL", msg)
+	log("INFO", format, args...)
```
Comment (krousey) [inline test/e2e/util.go]: So before, this was just calling with msg. Please change that back too.

File: test/e2e/util.go [hunk 5]
```diff
@@ -307,23 +307,23 @@ func nowStamp() string {
...
 	Fail(nowStamp()+": "+msg, 1)
 }
 
 func Skipf(format string, args ...interface{}) {
 	msg := fmt.Sprintf(format, args...)
-	logf("SKIP", msg)
+	log("INFO", format, args...)
```
Comment (krousey) [inline test/e2e/util.go]: Same here.

File: test/e2e/util.go (hunk 4 above)
Comment (goltermann) [inline test/e2e/util.go]: Fixed both.

**Comment by jdef** [comment]
mesos changes LGTM

File: pkg/api/v1/types.go [hunk 6]
```diff
@@ -2047,8 +2047,10 @@ type NodeSpec struct {
 
 // DaemonEndpoint contains information about a single Daemon endpoint.
 type DaemonEndpoint struct {
-	// Port number of the given endpoint.
-	Port int32 `json:port`
+	// Port number of the given endpoint. The tag was not properly in quotes in
+	// earlier releases, so it must be uppercased for backwards compat (since it
+	// was falling back to var name of 'Port').
```
Comment (krousey) [inline pkg/api/v1/types.go]: I'm not sure we want the comment showing up in the docs. @bgrant0607 Should he put  it in a separate comment block above this with a blank line between?

File: pkg/api/types.go [hunk 7]
```diff
@@ -1672,7 +1672,7 @@ type NodeSpec struct {
 // DaemonEndpoint contains information about a single Daemon endpoint.
 type DaemonEndpoint struct {
 	// Port number of the given endpoint.
-	Port int `json:port`
+	Port int `json:"port"`
```
Comment (krousey) [inline pkg/api/types.go]: @bgrant0607 Should this be `"Port"` as well?

Comment (goltermann) [inline pkg/api/types.go]: @bgrant0607 Opinions on the last two comments from Kris?

Comment (bgrant0607) [inline pkg/api/types.go]: @goltermann @krousey 

Yes.

So long as pkg/api/types.go still contains json tags, they should not differ from the versioned API without good reason.

Please suggest a clarification (via PR) to:
https://github.com/kubernetes/kubernetes/blob/master/docs/devel/api_changes.md#edit-typesgo-1

File: pkg/api/v1/types.go (hunk 6 above)
Comment (bgrant0607) [inline pkg/api/v1/types.go]: Yes.

File: pkg/apis/componentconfig/types.go [hunk 8]
```diff
@@ -230,7 +230,7 @@ type KubeletConfiguration struct {
 	// be rejected.
 	LowDiskSpaceThresholdMB int `json:"lowDiskSpaceThresholdMB"`
 	// How frequently to calculate and cache volume disk usage for all pods
-	VolumeStatsAggPeriod unversioned.Duration `json:volumeStatsAggPeriod`
+	VolumeStatsAggPeriod unversioned.Duration `json:"volumeStatsAggPeriod"`
```
Comment (goltermann) [inline pkg/apis/componentconfig/types.go]: @krousey Does this need to be uppercased for backcompat too?

Comment (krousey) [inline pkg/apis/componentconfig/types.go]: For this one, I'm not sure. It doesn't seem like a normal API type, but if it's user visible, I would imagine so. Again, I defer to @bgrant0607 on these.

Comment (bgrant0607) [inline pkg/apis/componentconfig/types.go]: @mikedanese Is anyone using these types yet?

Clearly we need a checker for this.

Comment (krousey) [inline pkg/apis/componentconfig/types.go]: @bgrant0607 This is what @goltermann is working torwards... as soon as we get the current codebase go vet clean, he will turn it on by default.

Comment (mikedanese) [inline pkg/apis/componentconfig/types.go]: Change looks good to me. Lower case is consistent with the rest of the file so I would leave it. This is not used yet and IIUC json tags aren't necessary in the internal types.go so we may remove them later. Thanks!

**Comment by goltermann** [comment]
@k8s-bot test this issue: #23952

**Pull request merged**